c7n/filters/core.py

```diff
@@ -816,3 +818,107 @@ def get_resource_value(self, resource):
         if capture is None:  # regex didn't capture anything
             return None
         return capture.group(1)
+
+
+class ReduceFilter(ValueFilter):
+    """Generic reduce filter to group, sort, and limit your resources.
+
+    This example will select the longest running instance from each ASG,
+    hen randomly choose 10% of those, maxing at 15 total instances.
+
+    filters:
+      - "tag:aws:autoscaling:groupName": present
+      - type: reduce
+        group_by: "tag:aws:autoscaling:groupName"
+        sort_by: "LaunchTime"
+        order: asc
+        limit: 1
+      - type: reduce
```

Comment:
these make more sense in separate policy examples? also. these aren't quite full policies, we actually run policies in doc strings ( yaml restructured text code blocks) through schema validation in ci.

Reply:
sure. i'll split them and make sure they're complete.  I didn't realize that happened.. that's great.